app.py

```diff
@@ -31,46 +30,176 @@ def init_app(self):
         }
         
         self.engine.register_command("show_in_filesystem", self.show_in_filesystem, p)
-            
+
     def launch(self, path):
-        self.log_debug("Launching file system viewer for folder %s" % path)        
-        
-        # get the setting        
+        """
+        Given a path this method will choose the appropriate way to open that as a folder in the OS's default
+        file manager. Folders will be opened as is
+        :param path:
+        :return: None
+        """
+
+        if os.path.isfile(path):
+            self._launch_filemanager_for_file(path)
+        elif os.path.isdir(path):
+            self._launch_filemanager_for_folder(path)
+        else:
+            # possibly we have an image sequence and therefore its a symbolic path, instead check to see if
+            # the parent folder of the path is valid and try opening that.
+            # todo: The issue with this logic is that possibly it was a directory that just didn't exist,
+            # so we would just be gathering the next directory up,
+            # ideally need a better way to handle sequences.
+            parent_dir = os.path.dirname(path)
+            if parent_dir and os.path.isdir(parent_dir):
```

Comment:
when would parent dir not be a directory here? do we need the isdir check? maybe worth adding a comment

Reply:
I think your right we do not want the `isdir` check. `dirname`, is just a string operation as far as I can tell and doesn't actually check for the existence of the directory, so I thought I should check for its actual existence, however looking back over this, its probably better to let the `_launch_filemanager_for_folder` raise the error if it failed to open the directory.